### ISSUE TITLE
transparent background?

### ISSUE THREAD

**Issue by Astrantia** [hackerb9/lsix#52]
hello and thanks for the project.

is it possible to set the montaged gallery background to transparent color? I tried setting `montage "${onerow[@]}"  $imoptions -background none gif:- | convert - -background 'rgba(0,0,0,0)' sixel:-` but it doesn't seem to work. Does sixel not support viewing transparent background or what could be wrong?

**Comment by hackerb9** [comment]
The sixel format supports 1-bit transparency, but ImageMagick does not know how to do that yet. The best we can do until that bug is fixed is to detect the terminal's background color and use that. It's not a bad solution as it means we also get antialiased fonts and drop shadows. Does your terminal not report the correct background color? Or, are you using an image or transparent background for your terminal?

**Comment by Astrantia** [comment]
> The sixel format supports 1-bit transparency, but ImageMagick does not know how to do that yet. The best we can do until that bug is fixed is to detect the terminal's background color and use that. It's not a bad solution as it means we also get antialiased fonts and drop shadows. Does your terminal not report the correct background color? Or, are you using an image or transparent background in your terminal?

lsix is correctly detecting my terminal's background color, but i've set my terminal itself to be 70% transparent, so the background color ends up being a solid color. from what i gather from your reply, it is impossible to get a no-color background currently with imagemagick?

**Comment by hackerb9** [comment]
> from what i gather from your reply, it is impossible to get a no-color background currently with imagemagick?
Correct. And I'm not even sure that it would look good if it did work. You would lose antialiasing so it may look grotty.

> lsix is correctly detecting my terminal's background color, but i've set my terminal itself to be 70% transparent, so the background color ends up being a solid color. 

That makes sense. Can you please send a screenshot of the problem and let me know which terminal you're using? While there's no fix at the moment, I'd like to keep this bug open as a reminder that sixel transparency needs further work and to try some experiments to see if it is worth pursuing.

If you could also run `lsix > opaque.six` and attach that file to this bug, I'd appreciate it. I will see if I can make it transparent and how it looks when I do. Thanks!

**Comment by steven-omaha** [comment]
I just ran into the same problem. Here is how it looks for me.

Screenshot:
![screen](https://user-images.githubusercontent.com/35634100/155708450-4226d4e8-30a3-441e-a978-2dc53ddfcc19.png)

sixel file: https://gist.githubusercontent.com/steven-omaha/0a5e2f8927be5866d8b666e7efffd661/raw/520a08bac24b57d59418d938c9050c042ee75213/opaque.six
